simulator/Simulator.kt

```diff
@@ -119,22 +119,30 @@ open class Simulator(
         state.mem = mem
     }
 
-    fun run() {
+    fun run(plugins: List<SimulatorPlugin> = emptyList()) {
         while (!isDone()) {
-            step()
+            step(plugins)
         }
     }
 
-    fun runToBreakpoint() {
+    fun runToBreakpoint(plugins: List<SimulatorPlugin> = emptyList()) {
         if (!isDone()) {
             // We need to step past a breakpoint.
-            step()
+            step(plugins)
         }
         while (!isDone() && !atBreakpoint()) {
-            step()
+            step(plugins)
         }
     }
 
+    open fun step(plugins: List<SimulatorPlugin>): List<Diff> {

```

Comment:
This needs to be an optional list input as well.

Reply:
The step function is overloaded. If you call `step()` it will not call the function we are commenting on, but the version with zero elements further below.

The reason I did this was in order to factor the plugin specific code out of the regular step function which is pretty complex already as it is.

Reply:
oh I missed that. Didnt see the step below since it was hidden in github.